Adds information regarding linking models on the hub to a HF Space (#919)
huggingface/hub-docs: docs/hub/spaces-overview.md

<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/SSHClone2.png"/>
<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/SSHClone1.png"/>
</div>

## Linking Models on the Hub

> You can showcase all the models that your space links to by adding their model identifier in your Space's README. To do so you can define them under the `models` key. For example:

Reviewer: ```suggestion
You can showcase all the models and datasets that your Space links to by adding their identifier in your Space's README metadata. To do so, you can define them under the `models` and `datasets` keys. For example:
```

```
title: My lovely space
emoji: 🤗
colorFrom: blue
colorTo: green
sdk: docker
pinned: false
models:
- reach-vb/musicgen-large-fp16-endpoint
```

> 

Reviewer: Would you mind also adding something along the lines of the second point in https://huggingface.slack.com/archives/C02EMARJ65P/p1693213871117809?thread_ts=1693042670.654759&cid=C02EMARJ65P please?

Author: Good point! Done!

<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/SSHClone2.png"/>
<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/SSHClone1.png"/>
</div>

> ## Linking Models on the Hub

Reviewer 2: ```suggestion
## Linking Models and datasets on the Hub
```

> ## Linking Models and datasets on the Hub

Reviewer: ```suggestion
## Linking Models and Datasets on the Hub
```

> You can showcase all the models and datasets that your Space links to by adding their identifier in your Space's README metadata. To do so, you can define them under the `models` and `datasets` keys. In addition to listing the aretefacts in the README file, you can also record them in any `.py`, `.ini` or `.html` file as well. We'll parse it auto-magically! 

Reviewer 2: ```suggestion
You can showcase all the models and datasets that your Space links to by adding their identifier in your Space's README metadata. To do so, you can define them under the `models` and `datasets` keys. In addition to listing the artefacts in the README file, you can also record them in any `.py`, `.ini` or `.html` file as well. We'll parse it auto-magically! 
```

> Here's an example linking two models with a space:

Reviewer 2: ```suggestion
Here's an example linking two models from a space:
```
